src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -9,6 +9,174 @@ namespace Microsoft.ML.Runtime.Internal.CpuMath
 {
     public static partial class CpuMathUtils
     {
+        public static void MatTimesSrc(bool tran, bool add, AlignedArray mat, AlignedArray src, AlignedArray dst, int crun)
+        {
+            Contracts.Assert(mat.Size == dst.Size * src.Size);
+
+            if (Sse.IsSupported)
+            {
+                if (!tran)
+                {
+                    Contracts.Assert(0 <= crun && crun <= dst.Size);
```

Comment:
This feels odd to me, it feels more readable to do  `crun >= 0` -- also, you could do a common Contracts.Assert for that, it seems like in all the blocks you're doing `0 <= crun`

Reply:
It is also generally useful to split `&&` asserts into two separate asserts, as it allows easy diagnosis on failure.

Reply:
Thank you for your feedback!  Tanner also raised similar concerns about `Contracts.Asserts` like `count > 0`.  I have adopted these changes and documented them in https://github.com/briancylui/machinelearning/issues/2.